vmware-tanzu/cloud-native-security-inspector
# Data Exporter (Episode 2): Add docker file and K8s yaml file for the Data Exporter and add e2e test

Fixes ISSUE #161 

## Description
This is the part 2 implementation of the approved [proposal](https://github.com/vmware-tanzu/cloud-native-security-inspector/blob/develop-0.3/docs/design/data-exporter-phase-1.md)

Finished works:
Part 1: Data Exporter (Episode 1): Implement the Data Exporter Server and add unit test for it #177 

Next:
Part 3: Modify the 3 scanners to let them send report to Data Exporter
Part 4: Add a doc for developers about how to extend Data Exporter with more consumers


## Tests
### Manual Test
1. After change, installed the new narrows.
2. Verified that the exporter sevice and pod are up:
```
01:51:05 root@example cloud-native-security-inspector ±|exporter-on-cloud|→ kg all -n cnsi-system | grep exporter | awk '{print $1}'
pod/cnsi-exporter-69c786c9f-bcrs7
service/cnsi-exporter-service
deployment.apps/cnsi-exporter
replicaset.apps/cnsi-exporter-69c786c9f
```

3. Port forward the exporter to the node, and call the API of exporter to forward a demo report:
Note that the opensearch name is {service name+IP}

![Screenshot 2023-02-27 at 14 52 37](https://user-images.githubusercontent.com/15973672/221494404-0b9b960e-b07a-404f-831c-44526c38c40c.png)

4. Verified that the log of exporter looks good:
```
01:40:50 root@example ~ → klon cnsi-system cnsi-exporter-69c786c9f-bcrs7
2023-02-27T06:33:38Z [INFO] [/workspace/main.go:12]: data exporter is up and listening on port 6780
2023-02-27T06:42:57Z [INFO] [/workspace/handlers.go:66]: the enabled outputs: [openSearch]
2023-02-27T06:43:01Z [INFO] [/workspace/outputs/client.go:132]: post successfully with response code: 201
```

5. Verified that the report shows up on the opensearch

![Screenshot 2023-02-27 at 14 56 49](https://user-images.githubusercontent.com/15973672/221495086-a5636cb6-c1fa-4d48-a6a2-a8870b8f7c60.png)

### E2E Test

For now I added case for verifying the existence of the exporter pod.

I will add functionality test when pluging the kubebench scanner into exporter. Because that will save some time.
